sdk/storage/Azure.Storage.DataMovement/src/CommitChunkHandler.cs

```diff
@@ -119,25 +114,37 @@ public async ValueTask DisposeAsync()
             DipsoseHandlers();
         }
 
-        public void DipsoseHandlers()
+        private void DipsoseHandlers()
         {
             if (_transferType == TransferType.Sequential)
             {
-                _commitBlockHandler -= QueueBlockEvent;
+                _commitBlockHandler -= SequentialBlockEvent;
             }
-            _commitBlockHandler -= CommitBlockEvent;
+            _commitBlockHandler -= ConcurrentBlockEvent;
         }
 
-        private async Task CommitBlockEvent(StageChunkEventArgs args)
+        private async Task ConcurrentBlockEvent(StageChunkEventArgs args)
         {
-            if (args.Success)
+            try
             {
-                // Let's add to the channel, and our notifier will handle the chunks.
-                await _stageChunkChannel.Writer.WriteAsync(args, _cancellationToken).ConfigureAwait(false);
+                if (args.Success)
+                {
+                    // Let's add to the channel, and our notifier will handle the chunks.
+                    await _stageChunkChannel.Writer.WriteAsync(args, _cancellationToken).ConfigureAwait(false);
+                }
+                else
+                {
+                    // Log an unexpected error since it came back unsuccessful
+                    await _invokeFailedEventHandler(Errors.FailedChunkTransfer(
```

Comment:
nit: Since this is inside a try/catch, did we want to throw instead? If so, this would apply to this same pattern in other places/files in this review too.